types/bluebird/index.d.ts

```diff
@@ -37,25 +37,32 @@
 
 type CatchFilter<E> = (new (...args: any[]) => E) | ((error: E) => boolean) | (object & E);
 type IterableItem<R> = R extends Iterable<infer U> ? U : never;
-type IterableOrNever<R> = R extends Iterable<any> ? R : never;
+type IterableOrNever<R> = Extract<R, Iterable<any>>;
+type Resolvable<R> = R | PromiseLike<R>;
+type ExtractFunction<T> = Extract<T, (...args: any[]) => any>;
+type IterateFunction<T, R> = (item: T, index: number, arrayLength: number) => Resolvable<R>;
+type IteraterItemIterateFunction<I, R> = IterateFunction<IterableItem<I>, R>;
```

Comment:
I see you're using `IteraterItemIterateFunction` for `map()`, `filter()`, `each()` and `mapSeries()`, but... As far as I can see all these are wrongly typed.

First of all you must keep in mind that these 4 methods are aliases of their equally named static counterparts (e.g. `static map(...);`)! This means the typings for e.g. `map()` and `static map()` must be identical too, which is not the case in your PR anymore.

Furthermore these functions do not return iterable items but rather regular (single) items.

As far as I can see these 4 functions must use `IterateFunction` (or `MapperFunction` as I suggested) instead of `IteraterItemIterateFunction`.

Reply:
You are right. I will follow the suggestions of this review at the next commit.